tests/test_clan_remaning_war_attacks.py

```diff
@@ -0,0 +1,184 @@
+                    {"tag": "D"},
+                    {"tag": "E"}
+                ],
+                191
+            ),
+            # Test 2 - Since D left, deckUsedToday for player D is 4 -> (200 - (1+2+4+4))
+            (
+                [
+                    {"tag": "B"},
+                    {"tag": "A"},
+                    {"tag": "C"}
+                ],
+                189
+            ),
+            # Test 3 - Since E left and not used any decks, decks remaining is -> (200 - (1+2+4+2))
+            (
+                [
+                    {"tag": "A"},
+                    {"tag": "B"},
+                    {"tag": "C"},
+                    {"tag": "D"}
+                ],
+                191
+            )
+        ]
+    )
+    def test_get_decks_remaining(self, members_list, expected_decks_remaining):
+        service = ClashRoyaleService()
+        participants = [
+            {"tag": "A", "decksUsedToday": 1},
+            {"tag": "B", "decksUsedToday": 2},
+            {"tag": "C", "decksUsedToday": 4},
+            {"tag": "D", "decksUsedToday": 2},
+            {"tag": "E", "decksUsedToday": 0}
+        ]
+        assert service._get_decks_remaining(
+            participants, members_list) == expected_decks_remaining
+
+    def test_clan_remaining_war_attacks(self):
+        service = ClashRoyaleService()
+        clan_tag = "#9GULPJ9L"
+        other_clan_tag = {
+            "PERU CB": "#P8UGJRGY",
+            "A Lithuania": "#UQ8VVR",
+            "Red Hot Chile": "#PR2U99V",
+            "Royal Crusade": "#2ULLJPUC",
+            "AUSCLAN": "#9GULPJ9L"
+        }
+        # Mock API responses and certain function calls
+        when(service.clash_royale_client).get_current_river_race(clan_tag).thenReturn(
+            clash_royale_client_currentriverrace.CURRENT_RIVER_RACE_API_RESPONSE
+        )
+        # MOCK CLAN INFO RESPONSE FOR PERU CB
+        when(service.clash_royale_client).get_clan_info(other_clan_tag["PERU CB"]).thenReturn(
+            clash_royale_client_responses.PER_CLAN_INFO_API_RESPONSE
+        )
+        # MOCK CLAN INFO RESPONSE FOR A Lithuania
+        when(service.clash_royale_client).get_clan_info(other_clan_tag["A Lithuania"]).thenReturn(
+            clash_royale_client_responses.LIT_CLAN_INFO_API_RESPONSE
+        )
+        # MOCK CLAN INFO RESPONSE FOR Red Hot Chile
+        when(service.clash_royale_client).get_clan_info(other_clan_tag["Red Hot Chile"]).thenReturn(
+            clash_royale_client_responses.RED_CLAN_INFO_API_RESPONSE
+        )
+        # MOCK CLAN INFO RESPONSE FOR Royal Crusade
+        when(service.clash_royale_client).get_clan_info(other_clan_tag["Royal Crusade"]).thenReturn(
+            clash_royale_client_responses.ROY_CLAN_INFO_API_RESPONSE
+        )
+        # MOCK CLAN INFO RESPONSE FOR AUSCLAN
+        when(service.clash_royale_client).get_clan_info(clan_tag).thenReturn(
+            clash_royale_client_responses.CLAN_INFO_API_RESPONSE
+        )
```

Comment:
This isn't needed anymore right? Can comment out since multiprocessing messes up mocking for these functions 😅 
Damn python man, I don't have these problems in other programming languages